tests/data-fetch-mixin.js

```diff
@@ -197,8 +197,48 @@ describe("DataFetch mixin", function() {
     var prevCallCount = fakeComponent.setState.callCount;
 
     var onError = $.ajax.args[0][0].error;
-    onError(null, 503, 'foobar');
+    onError({}, 503, 'foobar');
 
     expect(fakeComponent.setState.callCount).to.equal(prevCallCount);
-  })
+  });
+
+  it("should set isFetchingData to false in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.isFetchingData).to.equal(false);
+  });
+
+  it("should set dataError to null in initial state", function() {
+    var initialState = fakeComponent.getInitialState();
+
+    expect(initialState.dataError).to.equal(null);
+  });
+
+  it("should set dataError for a failed request", function() {
+    fakeComponent.props.dataUrl = 'foo';
+
+    fakeComponent.componentWillMount();
+
+    var onError = $.ajax.args[0][0].error;
+    onError({}, 404, 'foo');
+
+    var setStateArgs = fakeComponent.setState.lastCall.args[0];
+    expect(setStateArgs.dataError.statusText).to.equal(404);
+  });
+
+  it("should reset dataError before a new request", function() {
+    fakeComponent.props.dataUrl = 'foo';
+
+    fakeComponent.componentWillMount();
+
+    // force an error
+    var onError = $.ajax.args[0][0].error;
+    onError({}, 404, 'foo');
+
+    // force a new request that triggers the reset
+    fakeComponent.componentWillMount();
```

Comment:
A component will never have `componentWillMount` called twice in its lifetime. You can split this test into two:

- `"should reset dataError when refreshing data"` where you call `refreshData` after the error occurs
- `"should reset dataError when receiving new dataUrl"` where you call `componentWillReceiveProps` after changing dataUrl in props, after the error occurs